File: apps/ios/AccessMate/Screens/PassportScreen.swift
import SwiftUI

struct PassportScreen: View {
    @EnvironmentObject var stores: Stores
    @State private var editing = false

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 20) {
                header

                if stores.profile.profile.isEmpty {
                    CardSurface {
                        VStack(alignment: .leading, spacing: 12) {
                            Text("Set up your passport")
                                .font(.system(size: 22, weight: .bold, design: .serif))
                            Text("Staff can see your access needs at a glance. Takes about 90 seconds and stays on your device.")
                                .foregroundStyle(Theme.ink)
                            Button {
                                editing = true
                            } label: {
                                Label("Set up passport", systemImage: "person.text.rectangle.fill")
                                    .frame(maxWidth: .infinity, minHeight: 48)
                            }
                            .buttonStyle(.borderedProminent)
                            .controlSize(.large)
                            Text("You can skip anything and come back later.")
                                .font(.system(size: 13))
                                .foregroundStyle(Theme.inkMuted)
                        }
                    }
                } else {
                    if let m = stores.profile.profile.mobility {
                        CardSurface {
                            VStack(alignment: .leading, spacing: 8) {
                                SectionLabel(text: "Mobility")
                                ForEach(mobilityFacts(m), id: \.self) { Text($0) }
                            }
                        }
                    }
                    if let s = stores.profile.profile.sensory {
                        CardSurface {
                            VStack(alignment: .leading, spacing: 8) {
                                SectionLabel(text: "Sensory")
                                ForEach(sensoryFacts(s), id: \.self) { Text($0) }
                            }
                        }
                    }
                    if let c = stores.profile.profile.communication {
                        CardSurface {
                            VStack(alignment: .leading, spacing: 8) {
                                SectionLabel(text: "Communication")
                                ForEach(communicationFacts(c), id: \.self) { Text($0) }
                            }
                        }
                    }
                    if let n = stores.profile.profile.notes, !n.isEmpty {
                        CardSurface {
                            VStack(alignment: .leading, spacing: 8) {
                                SectionLabel(text: "Notes")
                                Text(n)
                            }
                        }
                    }
                    Button { editing = true } label: {
                        Label("Edit profile", systemImage: "pencil")
                            .frame(maxWidth: .infinity, minHeight: 48)
                    }
                    .buttonStyle(.bordered)
                    .controlSize(.large)
                }
            }
            .padding(20)
        }
        .background(Theme.paper)
        .navigationTitle("Passport")
        .sheet(isPresented: $editing) {
            NavigationStack {
                ProfileEditScreen()
            }
        }
    }

    private var header: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text("SHOW STAFF")
                .font(.system(size: 12, weight: .semibold))
                .tracking(1.4)
                .foregroundStyle(Theme.inkMuted)
            Text("Accessibility passport")
                .font(.system(size: 28, weight: .bold, design: .serif))
                .foregroundStyle(Theme.ink)
        }
    }

    private func mobilityFacts(_ m: ProfileMobility) -> [String] {
        switch m.wheelchairType {
        case "powered":          return ["Powered wheelchair"]
        case "manual":           return ["Manual wheelchair"]
        case "mobility-scooter": return ["Mobility scooter"]
        default:                 return []
        }
    }

    private func sensoryFacts(_ s: ProfileSensory) -> [String] {
        var out: [String] = []
        if s.isBlind == true { out.append("Blind") }
        if s.isLowVision == true { out.append("Low vision") }
        if s.isDeaf == true { out.append("Deaf") }
        if s.isHardOfHearing == true { out.append("Hard of hearing") }
        return out
    }

    private func communicationFacts(_ c: ProfileCommunication) -> [String] {
        var out: [String] = []
        if c.prefersBSL == true { out.append("Prefers British Sign Language") }
        if c.prefersWriting == true { out.append("Prefers written communication") }
        if c.needsExtraTime == true { out.append("Needs extra time when speaking or reading") }
        return out
    }
}

File: apps/android/app/src/main/java/com/accessmate/ui/screens/PassportScreen.kt
package com.accessmate.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.runtime.collectAsState
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.navigation.NavController
import com.accessmate.ui.appModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun PassportScreen(nav: NavController) {
    val vm = appModel()
    val profile by vm.profile.collectAsState()

    Scaffold(topBar = { TopAppBar(title = { Text("Accessibility passport") }) }) { padding ->
        Column(
            Modifier
                .padding(padding)
                .padding(16.dp)
                .verticalScroll(rememberScrollState()),
            verticalArrangement = Arrangement.spacedBy(16.dp),
        ) {
            if (profile.isEmpty) {
                Card {
                    Column(Modifier.padding(20.dp), verticalArrangement = Arrangement.spacedBy(12.dp)) {
                        Text("Set up your passport", style = MaterialTheme.typography.headlineSmall)
                        Text("Staff can see your access needs at a glance. Takes about 90 seconds and stays on your device.")
                        Button(
                            onClick = { nav.navigate("profile/edit") },
                            modifier = Modifier.fillMaxWidth(),
                        ) { Text("Set up passport") }
                        Text(
                            "You can skip anything and come back later.",
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                        )
                    }
                }
            } else {
                profile.mobility?.let { m ->
                    Card {
                        Column(Modifier.padding(16.dp)) {
                            Text("MOBILITY", style = MaterialTheme.typography.labelSmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
                            when (m.wheelchairType) {
                                "powered" -> Text("Powered wheelchair")
                                "manual" -> Text("Manual wheelchair")
                                "mobility-scooter" -> Text("Mobility scooter")
                            }
                        }
                    }
                }
                profile.sensory?.let { s ->
                    Card {
                        Column(Modifier.padding(16.dp)) {
                            Text("SENSORY", style = MaterialTheme.typography.labelSmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
                            if (s.isBlind == true) Text("Blind")
                            if (s.isLowVision == true) Text("Low vision")
                            if (s.isDeaf == true) Text("Deaf")
                            if (s.isHardOfHearing == true) Text("Hard of hearing")
                        }
                    }
                }
                profile.communication?.let { c ->
                    Card {
                        Column(Modifier.padding(16.dp)) {
                            Text("COMMUNICATION", style = MaterialTheme.typography.labelSmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
                            if (c.prefersBSL == true) Text("Prefers British Sign Language")
                            if (c.prefersWriting == true) Text("Prefers written communication")
                            if (c.needsExtraTime == true) Text("Needs extra time")
                        }
                    }
                }
                profile.notes?.takeIf { it.isNotBlank() }?.let { n ->
                    Card {
                        Column(Modifier.padding(16.dp)) {
                            Text("NOTES", style = MaterialTheme.typography.labelSmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
                            Text(n)
                        }
                    }
                }
                OutlinedButton(
                    onClick = { nav.navigate("profile/edit") },
                    modifier = Modifier.fillMaxWidth(),
                ) { Text("Edit profile") }
            }
        }
    }
}
